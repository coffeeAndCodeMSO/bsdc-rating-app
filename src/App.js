import React from 'react'
import style from './App.scss'

const App = () =>
  <h1 style={style.body}>
    Hello World!
  </h1>

export default App
