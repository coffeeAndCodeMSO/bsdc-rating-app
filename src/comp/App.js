import React from 'react'
import Counter from './Counter.js'
import style from '../css/App.scss'
import { createStore } from 'redux'
import counter from '../reducers'

const store = createStore(counter)

const App = () =>
  <div>
    <h1 style={style.body}>
      Hello to the World!
    </h1>
    <Counter
      value={store.getState()}
      onIncrement={() => store.dispatch({ type: 'INCREMENT' })}
      onDecrement={() => store.dispatch({ type: 'DECREMENT' })}
    />,
  </div>

store.subscribe(App)

export default App
