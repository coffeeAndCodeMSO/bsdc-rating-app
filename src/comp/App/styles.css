html, body {
  height: 100%;
  width:100%;
  background: #f2f2f2;
  overflow-x: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
}
