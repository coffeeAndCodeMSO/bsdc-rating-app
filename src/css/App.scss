$font-stack: Helvetica, sans-serif;
$primary-color: blue;
$flex-box: flex;

body {
  font: 100% $font-stack;
  color: $primary-color;
}

flexContainer {
  display: $flex-box;
}
