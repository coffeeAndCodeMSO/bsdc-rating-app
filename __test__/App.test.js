import React from 'react'
import App from '../src/comp/App'
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('main test block', () => {
  test('match snapshot', () => {
      const wrapper = shallow(
          <App />
      );
      expect(wrapper).toMatchSnapshot();
  });
})
