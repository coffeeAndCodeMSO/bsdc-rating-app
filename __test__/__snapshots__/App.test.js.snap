// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main test block match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          style={undefined}
        >
          Hello to the World!
        </h1>,
        <Counter
          onDecrement={[Function]}
          onIncrement={[Function]}
          value={0}
        />,
        ",",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello to the World!",
          "style": undefined,
        },
        "ref": null,
        "rendered": "Hello to the World!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onDecrement": [Function],
          "onIncrement": [Function],
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      ",",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={undefined}
          >
            Hello to the World!
          </h1>,
          <Counter
            onDecrement={[Function]}
            onIncrement={[Function]}
            value={0}
          />,
          ",",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello to the World!",
            "style": undefined,
          },
          "ref": null,
          "rendered": "Hello to the World!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onDecrement": [Function],
            "onIncrement": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        ",",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
